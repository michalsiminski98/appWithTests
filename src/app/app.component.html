<app-add-form></app-add-form>
<app-hide-button>SHOW / HIDE</app-hide-button>
<app-best-quote [title]= "titleForBest"></app-best-quote>
<app-quote-list [title] = "titleForList"></app-quote-list>
<app-worst-list [title] = "titleForWorst"></app-worst-list>