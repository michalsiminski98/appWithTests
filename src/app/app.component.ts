import { Component } from '@angular/core';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.css']
})
export class AppComponent {
  titleForBest: string = "Best Quote:"
  titleForList: string = "Quotes list:";
  titleForWorst: string = "Worst Quotes:";

  public isSubmitting: boolean = false;

  showForm = false;

  onShowForm(): void{
    this.showForm = !this.showForm;
  }

  adding(a:number, b:number): number{
    return a + b
  }
}

