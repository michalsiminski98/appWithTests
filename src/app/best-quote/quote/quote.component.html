<h3>{{author}}</h3>
<p>{{quotation}}</p>
<span>{{rate}}</span>
