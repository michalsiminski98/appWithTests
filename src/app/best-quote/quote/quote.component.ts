import { Component, Input } from '@angular/core';

@Component({
  selector: 'app-quote',
  templateUrl: './quote.component.html',
  styleUrls: ['./quote.component.css']
})
export class QuoteComponent {

  @Input() author?: string;
  @Input() quotation?: string;
  @Input() rate?: number;

  constructor() { }

}
