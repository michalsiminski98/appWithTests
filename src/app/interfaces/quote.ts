export interface Quote {
  id: number
  author: string;
  quotation: string;
  rate: number;
}