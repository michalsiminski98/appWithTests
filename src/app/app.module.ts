import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';

import { AppComponent } from './app.component';
import { HideButtonComponent } from './hide-button/hide-button.component';
import { AddFormComponent } from './add-form/add-form.component';
import { QuoteListComponent } from './quote-list/quote-list.component';
import { BestQuoteComponent } from './best-quote/best-quote.component';
import { WorstListComponent } from './worst-list/worst-list.component';
import { FormsModule } from '@angular/forms';

@NgModule({
  declarations: [
    AppComponent,
    HideButtonComponent,
    AddFormComponent,
    QuoteListComponent,
    BestQuoteComponent,
    WorstListComponent
  ],
  imports: [
    BrowserModule,
    FormsModule
  ],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule { }
