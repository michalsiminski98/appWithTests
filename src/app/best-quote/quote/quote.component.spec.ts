import { Shallow } from "shallow-render"
import { AppModule } from "src/app/app.module";
import { QuoteComponent } from "./quote.component"

describe('QuoteComponent', () =>{
  let shallow: Shallow<QuoteComponent>;

  beforeEach(() => {
    shallow = new Shallow(
      QuoteComponent,
      AppModule
    )
  });

  it('should match snapshot', async () => {
    const{instance} = await shallow.render();
    expect(instance).toMatchSnapshot();
  })
});
