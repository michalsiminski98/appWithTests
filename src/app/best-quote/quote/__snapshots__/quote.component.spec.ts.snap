// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuoteComponent should match snapshot 1`] = `QuoteComponent {}`;
